ignite-worker/src/main/java/org/opennms/poc/ignite/worker/workflows/impl/WorkflowExecutorLocalMonitorServiceImpl.java

```diff
@@ -79,12 +82,20 @@ public void cancel() {
 // Setup Internals
 //----------------------------------------
 
-    private ServiceMonitor lookupMonitor(Workflow workflow) {
+    private Optional<ServiceMonitor> lookupMonitor(Workflow workflow) {
         String pluginName = workflow.getPluginName();
 
-        Optional<ServiceMonitor> result = OsgiServiceHolder.getMonitor(pluginName);
+        Optional<ServiceMonitorManager> result = OsgiServiceHolder.getMonitorManager(pluginName);
 
-        return result.orElse(null);
+        if (result.isPresent()) {
+            ServiceMonitorManager foundMonitorManager = result.get();
+
+            PluginConfigInjector.injectConfigs(foundMonitorManager, workflow.getFieldConfigMeta());
```

Comment:
The `fieldConfigMeta` part should/may come from monitor manager.

Reply:
It actually does, though the loop I diagramed earlier. WE just need to resolve the issue of how to transmit it back as mentioned in other comments/feedback.